internal/ast/ast.go

```diff
@@ -273,25 +273,59 @@ type (
 		Cond Expr       // While loop condition.
 		Body *BlockStmt // Body of while.
 	}
+
+	// SwitchStatement node represents a switch statement
+	SwitchStatement struct {
+		Position
+		Value Expr       // Value to switch case
+		Body  *BlockStmt // Switch body
+	}
+
+	// SwitchCase node represents a single switch case
+	SwitchCase struct {
+		Position
+		Value Expr   // Condition to check to enter this case
+		Body  []Stmt // Switch case body
```

Comment:
I think that should be good add `or nil` at the end of the doc since a body of switch case can be nil.

Reply:
Done!